WarshipGame/Assets/Scripts/Gameplay/Grid/HexGrid.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// This class contains all the data of the grid, position of tiles [Vector2Int Coordinates]
+/// </summary>
+public class HexGrid : MonoBehaviour
+{
+    private Dictionary<Vector2Int, HexData> _hexTileDict = new Dictionary<Vector2Int, HexData>();
+    private Dictionary<Vector2Int, List<Vector2Int>> _hexTileNeighboursDict = new Dictionary<Vector2Int, List<Vector2Int>>();
+
+    private HexCoordinates _hexCoords;
+    
+    
+    //fills dictionary with hextiles
```

Comment:
fills needs a white space on the left and be capitalized
hextiles needs to be written in camelCase